src/main/java/org/reaktivity/nukleus/oauth/internal/stream/OAuthProxyFactory.java

```diff
@@ -510,12 +561,21 @@ private OAuthProxy(
             this.grant = Objects.requireNonNull(grant);
             this.isCorsPreflight = isCorsPreflight;
 
-            if (expiresAtMillis != EXPIRES_NEVER)
+//            final boolean canChallenge = Capabilities.canChallenge(capabilities);
+            final long delay;
+            if (challengeDelta > 0)
+            {
+                delay = expiresAtMillis - challengeDelta;
+
+                this.grantValidationFuture = executor.schedule(delay, MILLISECONDS, targetRouteId, targetStreamId,
+                        GRANT_VALIDATION_SIGNAL);
+            }
+            else if (expiresAtMillis != EXPIRES_NEVER)
             {
-                final long delay = expiresAtMillis - System.currentTimeMillis();
+                delay = expiresAtMillis - System.currentTimeMillis();
 
-                this.expiryFuture = executor.schedule(delay, MILLISECONDS, targetRouteId, targetStreamId, TOKEN_EXPIRED_SIGNAL);
-                this.grant.acquire();
+                this.grantValidationFuture = executor.schedule(delay, MILLISECONDS, targetRouteId, targetStreamId,
+                        GRANT_VALIDATION_SIGNAL);
```

Comment:
Please restructure this code to avoid the duplicate `executor.schedule` line.

Also I think the logic may not be quite right because `expiresAtMillis - challengeDelta` would be an absolute time, not a relative delay.
```
if (expiresAtMillis != EXPIRES_NEVER)
{
    final long signalDelay = expiresAtMillis - System.currentTimeMillis() - challengeResponseTimeout;
    this.signalFuture = executor.schedule(delay, MILLISECONDS, targetRouteId, targetStreamId, GRANT_VALIDATION_SIGNAL);
}
```
This assumes `challengeResponseTimeout` is `0` when not configured.